x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/integration/DocumentAndFieldLevelSecurityTests.java

```diff
@@ -475,11 +473,12 @@ private static void assertExpectedFields(FieldCapabilitiesResponse fieldCapabili
 
     private static void assertExpectedFields(Map<String, GetFieldMappingsResponse.FieldMappingMetadata> actual,
                                             String... expectedFields) {
-        Set<String> builtInMetadataFields = IndicesModule.getBuiltInMetadataFields();
         Map<String, GetFieldMappingsResponse.FieldMappingMetadata> fields = new HashMap<>(actual);
-        for (String field : builtInMetadataFields) {
-            GetFieldMappingsResponse.FieldMappingMetadata fieldMappingMetadata = fields.remove(field);
-            assertNotNull(" expected field [" + field + "] not found", fieldMappingMetadata);
+        for (String field : actual.keySet()) {
+            // best effort to remove metadata fields
+            if (field.startsWith("_")) {
+                assertNotNull(" expected field [" + field + "] not found", fields.remove(field));
+            }
```

Comment:
I wonder if `IndicesModule.getBuiltInMetadataFields` should be removed, or alternatively, improved so that it accounts for plugins like "_tier"?

Reply:
It is still used in a deprecated method called `MapperService#isMetadataFieldStatic`. I think it can be removed too but that will require a follow up. 
Regarding the plugins like `_tier`, the right method to use is `SearchExecutionContext#isMetadataField` which takes into account **all** metadata fields.